src/v/kafka/server/handlers/describe_transactions.cc

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "kafka/server/handlers/describe_transactions.h"
+
+#include "cluster/errc.h"
+#include "cluster/tx_gateway_frontend.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/protocol/schemata/describe_transactions_request.h"
+#include "kafka/protocol/types.h"
+#include "kafka/server/handlers/details/security.h"
+#include "kafka/server/request_context.h"
+#include "kafka/server/response.h"
+#include "model/fundamental.h"
+#include "model/namespace.h"
+#include "resource_mgmt/io_priority.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/when_all.hh>
+
+#include <chrono>
+#include <unordered_map>
+
+namespace kafka {
+
+namespace {
+
+ss::future<> fill_info_about_tx(
+  cluster::tx_gateway_frontend& tx_frontend,
+  describe_transactions_response& response,
+  kafka::transactional_id tx_id) {
+    describe_transaction_state tx_info_resp;
+    tx_info_resp.transactional_id = tx_id;
+
+    auto tx_info = co_await tx_frontend.describe_tx(tx_id);
+    if (tx_info.has_value()) {
+        auto tx = tx_info.value();
+        tx_info_resp.producer_id = kafka::producer_id(tx.pid.id);
+        tx_info_resp.producer_epoch = tx.pid.get_epoch();
+        tx_info_resp.transaction_state = ss::sstring(tx.get_kafka_status());
+        tx_info_resp.transaction_timeout_ms = tx.get_timeout() / 1ms;
```

Comment:
probably trying to do a conversion to milliseconds. glad you flagged:

@VadimPlh looking more closely at this, `tx_info::get_timeout` returns a `rm_stm::clock_type::duration_type` which is a `ss::lowres_clock`. that value is then placed into a response that will be sent back to the client.

the problem with sending back anything that originates from a `ss::lowres_clock` is that this clock has no correspondence to wall-clock time (e.g. it is generally counting from the time since a node booted up or a process started). so telling a client that some time T is important to them is meaningless if it comes from a lowres_clock.

time with lowres_clocks is still counted accurately, so a duration _might_ be fine to return, but it should raise concerns for anyone reviewing. for instance, durations/timeouts etc... can be independent of a particular clock type. for instance, tx_info::get_timeout should probably be returning clock-less std::chrono::milliseconds to make it clear.

---

in any case, changing all of that might be a lot of work. but what you could do in the short term is document as comments these cases that are not ideal, like this case.

Reply:
Some notes after yesterday chat with @mmaslankaprv 
1) It is not` ss::lowres_clock` anymore. I changed it to `lowres_system_clock`. We did not store clock_type::timepoint on disk. So we can do it without a lot of problem.
2) `time_since_epoch` from lowres_system_clock can be easy cast to milliseconds. @mmaslankaprv showed me fun way to do it by using `/ 1ms`. We have it is out codebase.
3) We store on disk only duration, so I added test that duration from `ss::lowres_clock` can be easly cast to milliseconds. And now we do not have problem with snapshots after changes `ss::lowres_clock` to `lowres_system_clock`.

Reply:
`lowres_system_clock` does not work
So I return back to lowers_clock and strange translation to normal time

Reply:
I think we can solve our issues here by changing  `tx_info_resp.transaction_timeout_ms` to be generated as `std::chrono::milliseconds` by our `generator.py`. Then it will be simply:
```c++

  tx_info_resp.transaction_timeout_ms = std::duration_cast<std::chrono::milliseconds>(tx.get_timeout());
```